lib/internal/modules/esm/loader.js

```diff
@@ -370,23 +373,16 @@ class ModuleLoader {
     const loadResult = defaultLoadSync(url, { format, importAttributes });
     const {
       format: finalFormat,
-      responseURL,
       source,
     } = loadResult;
 
-    this.validateLoadResult(url, finalFormat);
     if (finalFormat === 'wasm') {
       assert.fail('WASM is currently unsupported by require(esm)');
     }
 
-    const translator = getTranslators().get(finalFormat);
-    if (!translator) {
-      throw new ERR_UNKNOWN_MODULE_FORMAT(finalFormat, responseURL);
-    }
-
     const isMain = (parentURL === undefined);
-    const wrap = FunctionPrototypeCall(translator, this, responseURL, source, isMain);
-    assert(wrap instanceof ModuleWrap);  // No asynchronous translators should be called.
+    const wrap = this.#translate(url, finalFormat, source, isMain);
+    assert(wrap instanceof ModuleWrap, `Translator used for require(${url}) should not be async`);
```

Comment:
The assertions for translators could be removed if we just switch to `compiled = new WebAssembly.Module(source)` in the wasm translator, though I am not sure if anyone still wants to keep the imported wasm asynchronously compiled (import 'wasm' is not really all that useful after https://github.com/WebAssembly/esm-integration/blob/main/proposals/esm-integration/README.md is shipped by V8 anyway)